prototype_source/quantization_in_pytorch_2_0_export_tutorial.rst

```diff
@@ -0,0 +1,350 @@
+   output node; ``_annotated`` field indicates if this node has already been annotated by quantizer.
+   In this example, we will create the ``QuantizationAnnotation`` object with the ``QuantizationSpec`` objects
+   created in above step 2 for two inputs and one output of ``add`` node.
+
+::
+
+    input_qspec_map = {}
+    input_act0 = add_node.args[0]
+    input_qspec_map[input_act0] = input_act_qspec
+
+    input_act1 = add_node.args[1]
+    input_qspec_map[input_act1] = input_act_qspec
+         
+    add_node.meta["quantization_annotation"] = QuantizationAnnotation(
+        input_qspec_map=input_qspec_map,
+        output_qspec=output_act_qspec,
+        _annotated=True,
+    )
+
+After we annotate the ``add`` node like this, in the following up quantization flow, ``HistogramObserver`` will
+be inserted at its two input nodes and one output node in prepare phase. And ``HistogramObserver`` will be substituted with
+``quantize`` node and ``dequantize`` node in the convert phase.
+
+2. Annotate sharing qparams operators
+--------------------------------------------------------
+
+It is natural that users want to annotate a quantized model where quantization
+parameters can be shared among some tensors explicitly. Two typical use cases are:
+
+-  Example 1: One example is for ``add`` where having both inputs sharing quantization
+   parameters makes operator implementation much easier. Without using of
+   `SharedQuantizationSpec <https://github.com/pytorch/pytorch/blob/1ca2e993af6fa6934fca35da6970308ce227ddc7/torch/ao/quantization/_pt2e/quantizer/quantizer.py#L90>`__,
+   we must annotate ``add`` as example in above section 1, in which two inputs of ``add``
+   has different quantization parameters.
+-  Example 2: Another example is that of sharing quantization parameters between inputs and output.
+   This typically results from operators such as ``maxpool``, ``average_pool``, ``concat`` etc.
+
+``SharedQuantizationSpec`` is designed for this use case to annotate tensors whose quantization
+parameters are shared with other tensors. Input of ``SharedQuantizationSpec`` is an ``EdgeOrNode`` object which 
+can be an input edge or an output value. 
+
+-  Input edge is the connection between input node and the node consuming the input,
+   so it's a ``Tuple[Node, Node]``.
+-  Output value is an FX ``Node``.
+
+Now, if we want to rewrite ``add`` annotation example with ``SharedQuantizationSpec`` to indicate
+two input tensors as sharing quantization parameters. We can define its ``QuantizationAnnotation``
+as this:
+
+-  Step 1: Identify the original floating point pattern in the FX graph. We can use the same
+   methods introduced in ``QuantizationSpec`` example to identify the ``add`` pattern.
+-  Step 2: Annotate input_act0 of ``add`` with ``QuantizationSpec``.
+-  Step 3: Create a ``SharedQuantizationSpec`` object with input edge defined as ``(input_act0, add_node)`` which means to
```

Comment:
THis is exactly why we need to preface this by saying that annotation API requires quantizer to annotate nodes or edges. Nodes are annotated via `output_qspec` and edges can be annotated via `input_qspec_map`. Later we will show in edge can also be explicitly annotated when quantizer wants to express intention of sharing quantization parameters or deriving them.